### PR TITLE
Issue -269 Fixed

### PR DESCRIPTION
**Pull request by zakriya057**

### CODE REVIEW

**Comment by miesi** [comment]
```
$ ndcli create layer3domain test type vrf rd 256:256
type name    properties comment
vrf  default rd:8560:1  
vrf  test    rd:256:256
```
This is the output of list layer3domains

Please correct to something like
```
$ ndcli create layer3domain test type vrf rd 256:256
$ ndcli list layer3domains
type name    properties comment
vrf  default rd:8560:1  
vrf  test    rd:256:256
```

**Comment by miesi** [comment]
Please remove dim-testsuite/SHOW, dim_ndcli_test.patch from PR

Please add a $ ndcli list containers layer3domain test 0.0.0.0/0 after you have created v4 and v6 container in l3d test.

File: dim-testsuite/t/container-create-1.t [hunk 1]
```diff
@@ -0,0 +1,18 @@
+$ ndcli create container 0.0.0.0/0
+ ERROR - A layer3domain is needed
+
+$ ndcli create container 0.0.0.0/0 layer3domain default
+ntainer 0.0.0.0/0 layer3domain default
+INFO - Creating container 0.0.0.0/0 in layer3domain defaul
```
Comment (zeromind) [inline dim-testsuite/t/container-create-1.t:6]:
```suggestion
INFO - Creating container 0.0.0.0/0 in layer3domain default
```

File: dim-testsuite/t/container-create-1.t [hunk 2]
```diff
@@ -0,0 +1,18 @@
+$ ndcli create container 0.0.0.0/0
+ ERROR - A layer3domain is needed
+
+$ ndcli create container 0.0.0.0/0 layer3domain default
+ntainer 0.0.0.0/0 layer3domain default
```
Comment (zeromind) [inline dim-testsuite/t/container-create-1.t:5]:
```suggestion
```

File: dim-testsuite/t/container-create-1.t [hunk 3]
```diff
@@ -0,0 +1,18 @@
+$ ndcli create container 0.0.0.0/0
```
Comment (zeromind) [inline dim-testsuite/t/container-create-1.t:1]:
```suggestion
$ ndcli create layer3domain test type vrf rd 256:256

$ ndcli create container 0.0.0.0/0
```

**Comment by zakriya057** [comment]
Accept the changes in test cases.

**Comment by zakriya057** [comment]
Accept the changes

**Review by zakriya057 — COMMENTED** [review]
accept the changes

File: dim-testsuite/t/group-modify-1.t [hunk 4]
```diff
@@ -15,7 +15,7 @@ $ ndcli modify user-group global-dns-admins add user user2
 $ ndcli list user-group global-dns-admins users
 username
 user1
-user2
+user2-test
```
Comment (zeromind) [inline dim-testsuite/t/group-modify-1.t:18]:
```suggestion
user2
```

File: dim-testsuite/t/container-list-4.t [hunk 5]
```diff
@@ -0,0 +1,61 @@
+
```
Comment (zeromind) [inline dim-testsuite/t/container-list-4.t:1]:
```suggestion
```

File: dim/dim/rpc.py [hunk 6]
```diff
@@ -419,13 +419,13 @@ def ipblock_create(self, block_str, attributes=None, status='Container', disallo
             raise InvalidParameterError('Use ippool_add_subnet to create subnets')
 
         def find_parent():
+
```
Comment (zeromind) [inline dim/dim/rpc.py:422]:
```suggestion
```

File: dim/dim/rpc.py [hunk 7]
```diff
@@ -419,13 +419,13 @@ def ipblock_create(self, block_str, attributes=None, status='Container', disallo
             raise InvalidParameterError('Use ippool_add_subnet to create subnets')
 
         def find_parent():
+        
             parents = Ipblock._ancestors_noparent_query(parse_ip(block_str), None) \
                 .filter_by(status=get_status('Container')).all()
             if parents and all([p.layer3domain == parents[0].layer3domain for p in parents]):
                 return parents[0].layer3domain
-
```
Comment (zeromind) [inline dim/dim/rpc.py:426]: Leave empty line

File: dim/dim/rpc.py [hunk 8]
```diff
@@ -852,7 +852,6 @@ def explore(block):
                     [explore(c) for c in children] +
                     [{'ip': f, 'status': 'Available'} for f in block.free_space])
             return item
-
```
Comment (zeromind) [inline dim/dim/rpc.py:855]: Leave empty line

**Review by zeromind — CHANGES_REQUESTED** [review]
Minor formatting adjustments are necessary.

File: dim-testsuite/t/group-modify-1.t
[the remark this answers is not in the document]
Comment (miesi) [reply dim-testsuite/t/group-modify-1.t:1]: @zakriya057 please remove changes to `group-modify-1.t` from this PR.

**Review by miesi — APPROVED** [review]
Please remove dim.cfg and group-modify-1.t from this PR then merge in master ok.

**Comment by zakriya057** [comment]
I have removed the file and reverted the change from group-modify-1.t Now see changes in 2 files and 2 new files of test cases added in this PR.

**Pull request merged**